### PR TITLE
Command line todo list task

### PR DESCRIPTION
**Pull request by mbohnearschi**
Write an command line application that helps you manage your tasks. The information can be stored in any prefered way. The program should be launched from the console with different attributes:

-a or --add		adds an item into the list
-l or --list		displays the list
-r or --remove		removes the last n in the list
-c or --clear		clears the list
-h or --help		shows list of available commands

### CODE REVIEW

**Comment by mbohnearschi** [comment]
Hey @vlebedeff , should work now :)

File: tasks.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'optparse'
+
+OptionParser.new do |parser|
+
+  parser.on('-a', '--add TASK', 'Add an item to the list.') do |new_task|
+    save_tasks(tasks.push(new_task))
```
Comment (vlebedeff) [inline tasks.rb]: Looks like you're doing extra work here - 
calling `#tasks` to opens a file and loads the lines into memory, push the new task to in-memory array and then rewriting the whole list of tasks.
You could just open the file in append mode and write a single line to it:
`File.open('tasks.dat', 'a') { |f| f.write(new_task) }`

File: tasks.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  parser.on('-r', '--remove N', 'Remove the last N from the list.') do |n|
+    original_tasks = tasks
+    n.to_i.times do
+      original_tasks.pop
+    end
+    save_tasks(original_tasks)
+  end
+
+  parser.on('-c', '--clear', 'Clear the list.') do
+    save_tasks([])
```
Comment (vlebedeff) [inline tasks.rb]: `File.truncate('tasks.dat', 0)` or `File.open('tasks.data', 'w') {}`

File: tasks.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  parser.on('-l', '--list', 'Display the list.') do
+    tasks.each { |task| puts task }
+  end
+
+  parser.on('-r', '--remove N', 'Remove the last N from the list.') do |n|
+    original_tasks = tasks
+    n.to_i.times do
+      original_tasks.pop
+    end
+    save_tasks(original_tasks)
```
Comment (vlebedeff) [inline tasks.rb]: You could use [array slicing](https://apidock.com/ruby/Array/slice) with negative values of specify offset from the end. Something like:
```ruby
    remaining = File.readlines('tasks.dat')[0..-n.to_i]
    File.write('tasks.dat', remaining.join("\n"))
```

**Pull request merged**